projects/plugins/jetpack/extensions/blocks/donations/donations-view/edit.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * Internal dependencies
+ */
+import DonationsContext from '../common/context';
+import { ANNUAL_DONATION, MONTHLY_DONATION, ONE_TIME_DONATION } from '../common/constants';
+
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { useContext } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import { InnerBlocks } from '@wordpress/block-editor';
+
+const types = {
+	[ ONE_TIME_DONATION ]: {
+		heading: __( 'Make a one-time donation', 'jetpack' ),
+		cta: __( 'Donate', 'jetpack' ),
+	},
+	[ MONTHLY_DONATION ]: {
+		heading: __( 'Make a monthly donation', 'jetpack' ),
+		cta: __( 'Donate monthly', 'jetpack' ),
+	},
+	[ ANNUAL_DONATION ]: {
+		heading: __( 'Make a yearly donation', 'jetpack' ),
+		cta: __( 'Donate yearly', 'jetpack' ),
+	},
+};
+
+const Edit = props => {
+	const { attributes, setAttributes } = props;
+	const { type } = attributes;
+	const { heading, cta } = types[ type ];
+	const { activeTab, fallbackLinkUrl, products } = useContext( DonationsContext );
+	const planId = products[ type ];
+
+	setAttributes( { fallbackLinkUrl, planId } );
+
+	return (
+		<div className={ classNames( 'donations__view', { 'is-active': activeTab === type } ) }>
+			<InnerBlocks
+				templateLock={ true }
```

Comment:
I think this should be `'all'`.

Either way, locking the template is a good (probably _the only_) option for now.
Eventually, if Gutenberg adds proper support for per-block locking, we could instead:

* Unlock the templates.
* Only lock (i.e. prevent removing) the amounts and the donate button.

So that folks would have complete control over the content, whereas now they can only change the styles but are forced to have, for example, a title and a description.

The [API exists](https://github.com/WordPress/gutenberg/pull/32457), but I don't think is really supported.
Might be worth double-checking with Gutenberg folks though!

Reply:
The API is supported and is on production for WooCommerce blocks.